src/components/back-link-helper/back-link-helper.js

```diff
@@ -5,27 +5,20 @@ function BackLinkHelper($module, window, document) {
 }
 
 BackLinkHelper.prototype.init = function init() {
-  const isReferrerExternal = () => {
-    // If there is no referrer, consider it on the same domain
-    if (!this.document.referrer) {
-      return false;
-    }
-
-    try {
-      const currentDomain = new URL(this.window.location.href).hostname;
-      const referrerDomain = new URL(this.document.referrer).hostname;
-
-      // Check if the referrer is not the same as the current domain
-      return currentDomain !== referrerDomain;
-    } catch (err) {
-      return false;
-    }
-  };
-
   // do nothing if History API is absent
   if (this.window.history) {
-    // hide the backlink if the referrer is on a different domain
-    if (isReferrerExternal()) {
+    // eslint-disable-next-line max-len
+    /* TODO: It remains unclear whether a check for the same domain is necessary for security reasons.
+       There may be user research suggesting considerations regarding the visibility of the
+       back link on refresh.
+       Currently, a page refresh sets the referer to empty, leading to the back link being hidden
+       under our existing logic.
+     */
+    // eslint-disable-next-line max-len
+    const referrerNotOnSameDomain = () => this.document.referrer.indexOf(this.window.location.host) === -1;
```

Comment:
will this cause a js error if document.referrer is empty? I don't know if it will be "" or undefined, but if it can be undefined then indexOf might cause an error

Reply:
According to the Mozilla documentation, the referer will always be set to an empty string if it does not have a value.

https://developer.mozilla.org/en-US/docs/Web/API/Document/referrer

Reply:
that is good enough for me to be honest, can I use shows even old IE seems to support, so not sure why it was coded like that in old version, probably just bit more defensive coding than needed, https://caniuse.com/mdn-api_document_referrer